src/SAWScript/Crucible/MIR/ResolveSetupValue.hs

```diff
@@ -229,6 +255,28 @@ resolveSetupVal mcc env tyenv nameEnv val =
       let (fldShpAssn, valAssn) = Ctx.unzip fldAssn
       let tupleShp = TupleShape (Mir.TyTuple fldMirTys) fldMirTys fldShpAssn
       pure $ MIRVal tupleShp valAssn
+    MS.SetupSlice slice ->
+      case slice of
+        MirSetupSliceRaw{} ->
+          panic "resolveSetupVal" ["MirSetupSliceRaw not yet implemented"]
+        MirSetupSlice arrRef -> do
+          SetupSliceFromArray _elemTpr sliceShp refVal len <-
+            resolveSetupSliceFromArray bak arrRef
+          lenVal <- usizeBvLit sym len
+          pure $ MIRVal sliceShp (Ctx.Empty Ctx.:> RV refVal Ctx.:> RV lenVal)
+        MirSetupSliceRange arrRef start end -> do
+          unless (start <= end) $
+            fail $ "slice index starts at " ++ show start
+                ++ " but ends at " ++ show end
+          SetupSliceFromArray elemTpr sliceShp refVal0 len <-
+            resolveSetupSliceFromArray bak arrRef
+          unless (end < len) $
+            fail $ "range end index " ++ show end
+                ++ " out of range for slice of length " ++ show len
+          startBV <- usizeBvLit sym start
+          refVal1 <- Mir.mirRef_offsetIO bak iTypes elemTpr refVal0 startBV
+          lenVal <- usizeBvLit sym $ end - start
+          pure $ MIRVal sliceShp (Ctx.Empty Ctx.:> RV refVal1 Ctx.:> RV lenVal)
```

Comment:
This is where the magic happens, such as interfacing with the `crucible-mir` memory model and checking the start/end values of slice ranges.